stdlib/public/core/CMakeLists.txt

```diff
@@ -196,6 +196,13 @@ set(SWIFTLIB_ESSENTIAL_GYB_SOURCES
   UnsafeRawBufferPointer.swift.gyb
   )
 
+# Compile differentiable programming sources only if enabled.
+set(SWIFTLIB_DIFFERENTIABLE_PROGRAMMING_SOURCES)
+if(SWIFT_ENABLE_EXPERIMENTAL_DIFFERENTIABLE_PROGRAMMING)
+  # TODO: Add `_Differentiable` protocol.
```

Comment:
Note: https://github.com/apple/swift/pull/27511 will add the `_Differentiable` protocol - separate patch to facilitate review.